### DISCUSSION TITLE
How can I assign R and S configurations for docking poses using RDKit?

### DISCUSSION THREAD

**Discussion by n-vivek** [rdkit/rdkit#7988] [Q&A]
Hello everyone,

I’m working with docking results obtained from AutoDock Vina, and the poses are saved in PDBQT format. I want to assign the R and S configurations to these docking poses using RDKit. While I have a basic understanding of stereochemistry, I’m not very experienced with RDKit, and any guidance would be really helpful.

Specifically, I’m looking for advice on:
1. Using RDKit to calculate the R/S configuration for each pose.
2. How to handle multiple poses and save the results.

Any help or suggestions would be much appreciated. I’m eager to learn and improve my skills with RDKit.

Thank you!

**Comment by ptosco** [comment]
You can read each PDBQT result with [Chem.MolFromPDBFile](https://rdkit.org/docs/source/rdkit.Chem.rdmolfiles.html#rdkit.Chem.rdmolfiles.MolFromPDBFile). This will give you an RDKit molecule object with 3D coordinates. Make sure not to remove Hs. Note that the PDB format does not contain bond order information, which is necessary to correctly perceive stereochemistry. However, you know the bond orders of the ligand(s) you docked, so you can assign bond orders to your RDKit molecule using the SMILES representation of your docked ligands through [AllChem.AssignBondOrdersFromTemplate](https://rdkit.org/docs/source/rdkit.Chem.AllChem.html#rdkit.Chem.AllChem.AssignBondOrdersFromTemplate). Finally, you may perceive the stereochemistry of your 3D RDKit molecule with [Chem.AssignStereochemistryFrom3D](https://rdkit.org/docs/source/rdkit.Chem.rdmolops.html#rdkit.Chem.rdmolops.AssignStereochemistryFrom3D).

Here follows a complete example based on atenolol, assuming that PDB poses are in a directory called `atenolol_pdbs`.
Each pose is saved to a SDF file tagged with a `CHIRALITY` property, which in the case of racemic atenolol will be either `5S` or `5R`.
```
import os
from rdkit import Chem
from rdkit.Chem import AllChem

atenolol = Chem.MolFromSmiles("CC(C)NCC(COC1=CC=C(C=C1)CC(=O)N)O")

with Chem.SDWriter("/da/CADD/toscopa1/support/rdkit/atenolol_RS.sdf") as w:
    for pdb in sorted(os.listdir("atenolol_pdbs")):
        m = Chem.MolFromPDBFile(os.path.join("atenolol_pdbs", pdb), removeHs=False)
        assert m
        AllChem.AssignBondOrdersFromTemplate(atenolol, m)
        Chem.AssignStereochemistryFrom3D(m)
        chirality = ",".join(f"{idx}{cip}" for idx, cip in Chem.FindMolChiralCenters(m, useLegacyImplementation=False))
        m.SetProp("CHIRALITY", f"({chirality})")
        w.write(m)
```

**Reply by n-vivek** [reply]
Thank you so much for the advice, Mr. Tosco - really appreciate it! I’ll give your approach a try, as it seems promising.

I'm also working on a script to calculate the R and S configurations for an intermediate substrate that transforms into a product by interacting with an enzyme cofactor. In this case, the four atoms for the CIP rule won’t all be within the same molecule, with one derived from the cofactor. Therefore, I’m attempting to predict the stereochemical outcome by focusing on reaction geometry rather than solely on the current molecular configuration. It’s both challenging and exciting - let’s see how it unfolds!
